images_config/kickstart/kickstart-robot.sh

```diff
@@ -0,0 +1,465 @@
+        wget -O /etc/network/interfaces.backup https://raw.githubusercontent.com/ipa-bnm/setup_cob4/feature/xenial_unattended/cob-pcs/networkInterfacesSlave
+    fi
+
+    sed -i "s/eth0/$INTERFACE/g" /etc/network/interfaces.backup
+
+    echo "mv /etc/network/interfaces.backup /etc/network/interfaces && ifup -a && sed -i '/fixnet.sh/d' /etc/rc.local && rm -f /fixnet.sh" > /fixnet.sh
+    sed -i '$ibash /fixnet.sh' /etc/rc.local
+    systemctl restart networking
+}
+
+function SetupEtcHosts {
+    printHeader "SetupEtcHosts"
+    HOSTNAME=$(cat /etc/hostname)
+
+    sed -i "s/$HOSTNAME.wlrob.net\t//g" /etc/hosts
+
+    ROBOTNAME=$(echo ${HOSTNAME%-*})
+    ROBOT_NUM=$(echo ${ROBOTNAME##*-})
+
+    PC_LS=(
+    "10.4.${ROBOT_NUM}.41	${ROBOTNAME}-h1"
+    "10.4.${ROBOT_NUM}.31	${ROBOTNAME}-s1"
+    "10.4.${ROBOT_NUM}.23	${ROBOTNAME}-t3"
+    "10.4.${ROBOT_NUM}.22	${ROBOTNAME}-t2"
+    "10.4.${ROBOT_NUM}.21	${ROBOTNAME}-t1"
+    "10.4.${ROBOT_NUM}.11	${ROBOTNAME}-b1"
+    )
+
+    for ((i = 0; i < ${#PC_LS[@]}; i++))
+    do
+    if ! [[ ${PC_LS[$i]} == *${HOSTNAME}* ]]; then
+        sed -i "4i ${PC_LS[$i]}" /etc/hosts
+    fi
+    done
+
+    sed -i $'8 a \n' /etc/hosts
+}
+
+function InstallCandumpTools {
+    printHeader "InstallCandumpTools"
+    wget -O /usr/local/bin/socket_buffer.py https://raw.githubusercontent.com/ipa320/setup_cob4/master/scripts/socket_buffer.py
+    chmod +x /usr/local/bin/socket_buffer.py
+}
+
+#TODO: only newest NoMachine ?!?
+function InstallNoMachine {
+    printHeader "InstallNoMachine"
+    NOMACHINE_VERSION=6.0.66_2
+    wget -O /root/nomachine_${NOMACHINE_VERSION}_amd64.deb http://download.nomachine.com/download/6.0/Linux/nomachine_${NOMACHINE_VERSION}_amd64.deb
+    dpkg -i /root/nomachine_${NOMACHINE_VERSION}_amd64.deb
+}
+
+function InstallNetData {
+    printHeader "InstallNetData"
+    apt-get install zlib1g-dev uuid-dev libmnl-dev gcc make git autoconf autoconf-archive autogen automake pkg-config curl -y
+    git clone https://github.com/firehol/netdata.git --depth=1 /root/netdata
+    cd /root/netdata
+    yes | ./netdata-installer.sh
+    cd
+    rm -r /root/netdata
+}
+
+function InstallCobCommand {
+    printHeader "InstallCobCommand"
+    if [ "$INSTALL_TYPE" == "master" ]; then
+        wget -O /usr/sbin/cob-command https://raw.githubusercontent.com/ipa320/setup_cob4/master/scripts/cob-command
+        chmod +x /usr/sbin/cob-command
+        sh -c 'echo "%users ALL=NOPASSWD:/usr/sbin/cob-command"' | sed -i -e "\|%users ALL=NOPASSWD:/usr/sbin/cob-command|h; \${x;s|%users ALL=NOPASSWD:/usr/sbin/cob-command||;{g;t};a\\" -e "%users ALL=NOPASSWD:/usr/sbin/cob-command" -e "}" /etc/sudoers
+    fi
+}
+
+function RemoveModemanager {
+    printHeader "RemoveModemanager"
+    apt-get purge modemmanager -y
+}
+
+function DisableUpdatePopup {
+    printHeader "DisableUpdatePopup"
+    sed -i 's/Prompt\=lts/Prompt\=never/g' /etc/update-manager/release-upgrades
+}
+
+# TODO: fxm ;)
+# Not functional under xenial
```

Comment:
under trusty, this guaranteed that the slave pc's only booted with network set up...which was required to guarantee that nfs is running when logging into the mimic user....which was required for the GUI to be displayed properly via vlc

we need to find a solution for this under xenial, too!

Reply:
maybe `/etc/X11/Xsession.options` could be modified under `xenial` instead...

there is an option `failsafe-allowed` - that could be commented out...

Reply:
As far as i understood, the failsafe boot has nothing to do with the graphical failsafe screen (e.g. "starting low graphics mode - graphics driver problem". Failsafe in this case has something to do with the runlevel. Runlevel 1 is recovery/single-mode and the script for 14.04 prevents recovery/single-mode.
We should observe if we still need it.